crates/docs/src/validation.rs

```diff
@@ -0,0 +1,70 @@
+use regex::Regex;
+use std::{
+    env, fs, io,
+    path::{Path, PathBuf},
+};
+
+pub fn extract_matches_from_file(
+    file_path: &Path,
+    re: &Regex,
+) -> Result<Vec<String>, std::io::Error> {
+    let content = fs::read_to_string(file_path)?;
+    let matches: Vec<String> = re
+        .captures_iter(&content)
+        .filter_map(|caps| caps.get(1).map(|m| m.as_str().to_string()))
+        .collect();
```

Comment:
Nit but maybe this can be flattened & renamed for readability

```suggestion
    let matches: Vec<String> = re
        .captures_iter(&content)
        .filter_map(|captures| captures.get(1))
        .map(|m| m.as_str().to_string())
        .collect();
```